gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -239,7 +240,8 @@ private WriteObjectResponse doResumableUpload() throws IOException {
 
       // Wait for streaming RPC to become ready for upload.
       try {
-        responseObserver.ready.await();
+        // wait for 1 min for the channel to be ready. Else bail out
+        responseObserver.ready.await(60*1000, MILLISECONDS);
```

Comment:
We don't have telemetry data on the delay we observe. The issue itself is fairly rare to reproduce. This timeout can be related to http-open timeout because here we are trying to establish grpc stream I guess. We could use the same timeout or as you suggest create a new configurable parameter. 

The fact that it is an extremely rare scenario, makes me lean towards not making it a configurable parameter. When this case is hit, we do retry it internally, so it is not a user facing consequence.